docker/docker.mk

```diff
@@ -50,41 +51,55 @@ check-docker-env:
 ifeq ($(DAPR_REGISTRY),)
 	$(error DAPR_REGISTRY environment variable must be set)
 endif
-ifeq ($(DAPR_TAG),)
-	$(error DAPR_TAG environment variable must be set)
+ifeq ($(REL_VERSION),)
+	$(error REL_VERSION environment variable must be set)
+endif
+
+check-arch:
+ifeq ($(TARGET_OS),)
+	$(error TARGET_OS environment variable must be set)
+endif
+ifeq ($(TARGET_ARCH),)
+	$(error TARGET_ARCH environment variable must be set)
 endif
 
 # build docker image for linux
-docker-build: check-docker-env
+BIN_PATH=$(OUT_DIR)/$(TARGET_OS)_$(TARGET_ARCH)/release
+
+docker-build: check-docker-env check-arch
 	$(info Building $(DOCKER_IMAGE_TAG) docker image ...)
-	$(DOCKER) build --build-arg TARGETPLATFORM=linux/amd64 --build-arg PKG_FILES=* -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(OUT_DIR)/. -t $(DOCKER_IMAGE_TAG)
-	$(DOCKER) build --build-arg TARGETPLATFORM=linux/amd64 --build-arg PKG_FILES=daprd -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(OUT_DIR)/. -t $(DAPR_RUNTIME_DOCKER_IMAGE_TAG)
-	$(DOCKER) build --build-arg TARGETPLATFORM=linux/amd64 --build-arg PKG_FILES=placement -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(OUT_DIR)/. -t $(DAPR_PLACEMENT_DOCKER_IMAGE_TAG)
-	$(DOCKER) build --build-arg TARGETPLATFORM=linux/amd64 --build-arg PKG_FILES=sentry -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(OUT_DIR)/. -t $(DAPR_SENTRY_DOCKER_IMAGE_TAG)
+	$(DOCKER) build --build-arg PKG_FILES=* -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DOCKER_IMAGE_TAG)-$(TARGET_OS)-$(TARGET_ARCH)
+	$(DOCKER) build --build-arg PKG_FILES=daprd -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DAPR_RUNTIME_DOCKER_IMAGE_TAG)-$(TARGET_OS)-$(TARGET_ARCH)
+	$(DOCKER) build --build-arg PKG_FILES=placement -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DAPR_PLACEMENT_DOCKER_IMAGE_TAG)-$(TARGET_OS)-$(TARGET_ARCH)
+	$(DOCKER) build --build-arg PKG_FILES=sentry -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DAPR_SENTRY_DOCKER_IMAGE_TAG)-$(TARGET_OS)-$(TARGET_ARCH)
```

Comment:
this is used for e2e and perf test. please double-check if it break these tests.

Reply:
I think we are going to have to add TARGET_OS and TARGET_ARCH as mandatory e2e test parameters. I'm working on windows E2E tests, which are going to need to be run against windows enabled k8s clusters. We'll have to build the containers on a windows machine, thus we're going to have to parameterize our tests to run the windows specific bits.